### DISCUSSION TITLE
Slow computation

### DISCUSSION THREAD

**Discussion by adam-bielanski** [alicevision/Meshroom#1712] [Ideas]
Hi,
IS there any chance to speed up Meshroom in the 2022 release? Current version is painfully slow and blocks the computer completely for 2-3days.... sometimes longer.
A

**Comment by natowi** [comment]
This sounds like you are using an old computer or too many images. There is nothing you can do to improve the performance on your old hardware except reducing the quality, but in case of too many images this can be optimized. For most models, up to a few hundred images suffice. If you have a dataset of - lets say 1000+ images - you need to check if they are really necessary to reconstruct the model.

**Comment by adam-bielanski** [comment]
Hi,
It is not a hardware issue, but the simple comparison to some commercial software.

I operate on fairly new machine with RTX card, so I assume it should not have a performance problems. Unfortunately the 30+ hours of model meshing seems too long, whilst testing the same set of images on other software took up to 2-3 hours. 

On the other hand I admire the level of  precision I can achieve with meshroom. 


⁣Pobierz aplikację TypeApp dla systemu Android ​

W dniu 16 cze 2022, 19:59, o 19:59, użytkownik Simon ***@***.***> napisał:
 >This sounds like you are using an old computer or too many images.
>There is nothing you can do to improve the performance on your old
>hardware except reducing the quality, but in case of too many images
>this can be optimized. For most models, up to a few hundred images
>suffice. If you have a dataset of - lets say 1000+ images - you need to
>check if they are really necessary to reconstruct the model.
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/alicevision/meshroom/discussions/1712#discussioncomment-2966018
>You are receiving this because you authored the thread.
>
>Message ID:
***@***.***>

**Comment by adam-bielanski** [comment]
Just to be clear - I test meshroom on architectural tasks and I look at
this software from the practical, not academic point of view, therefore it
is always "hundreds" of images  (last time it was 1400). Quality may exceed
the commercial competitiors, but the timing is crucial if that software is
supposed to be anything else than purely academic testing of new ideas.

Please do not treat it as a complain, just an oppinion of the tester of
last 3 releases.

czw., 16 cze 2022 o 19:59 Simon ***@***.***> napisał(a):

**Reply by natowi** [reply]
AliceVision is in constant development, and there are numerous recent improvements (wip) - https://github.com/alicevision/AliceVision/pull/1296 and https://github.com/alicevision/AliceVision/pull/1236 for example
But full hardware utilization and code optimization is not an easy matter. Reducing computation time while maintaining high quality results is a special challenge.
Commercial software have significantly more funding and full time developers, so they spend all resources to improve the performance.

Who would use open-source software that is fast, but delivers crappy results? So it makes sense to focus on quality first and work on improving performance later :)
